dali/server/daserver.cpp

```diff
@@ -136,6 +139,199 @@ void usage(void)
     printf("--daemon|-d <instanceName>\t: run daemon as instance\n");
 }
 
+/* NB: Ideally this belongs within common/environment,
+ * however, that would introduce a circular dependency.
+ */
+static bool populateWhiteListFromEnvironment(IWhiteListHandler &whiteListHandler)
+{
+    Owned<IRemoteConnection> conn = querySDS().connect("/Environment", 0, 0, INFINITE);
+    assertex(conn);
+
+    // only ever expecting 1 DaliServerProcess and 1 WhiteList
+    IPropertyTree *whiteListTree = conn->queryRoot()->queryPropTree("Software/DaliServerProcess[1]/WhiteList[1]");
+    if (whiteListTree)
+    {
+        if (!whiteListTree->getPropBool("@enabled", true)) // on by default
+            return false;
+        // Default for now is to allow clients that send no role (legacy) to connect if their IP is whitelisted.
+        whiteListHandler.setAllowAnonRoles(whiteListTree->getPropBool("@allowAnonRoles", true));
+    }
+
+    std::unordered_map<std::string, std::string> machineMap;
+
+    auto populateMachineMap = [&machineMap](IPropertyTree &environment)
```

Comment:
I prefer the previous code structure - as members of a WhiteListClass (which could implement IWhiteLIstHandler).  Not sure what I think of use of lambda functions like this.  (Not used to it, may be a good idea, but is it more confusing than helpful?)

Reply:
The reason I changed it from a class to a single function,
was that I wanted the 'populate' to be single callback function and not part of the interface.

The implementation doesn't know about how to populate, but calls the supplied function to do so on demand.

(the lamda function could just be outside, but seems tidier to encapsulate the function here).